ood-test.osc.edu/apps/myjobs/templates/Simple_OMP_Job_Slurm/omp-hello.sh

```diff
@@ -0,0 +1,29 @@
+#!/bin/bash
+
+#SBATCH --job-name="Simple OpenMP job"
+#SBATCH -J ondemand/sys/myjobs/omp_job_slurm
+#SBATCH --output=omp-hello.out
+#SBATCH -t 00:01:00
+#SBATCH --nodes=1
+#SBATCH --exclusive
+
+# Load the OpenMPI module
+module load intel/19.0.5
+module load openmpi/4.0.3-hpcx
```

Comment:
I would not recommend using openmpi HPCX because HPCX versions of OpenMPI are tightly coupled to the OS version and thus we frequetly have to add/remove HPCX versions to match what OS version we are on.  Use mvapich2.

Reply:
If you must use openmpi, use the non-HPCX versions.

Reply:
When using `srun` this error happens:

```error while loading shared libraries: libiomp5.so: cannot open shared object file: No such file or directory```

Reply:
I'd have to see how exactly you compiled the code, which modules were loaded and how the job was executed because I took this directory and it works.

```
module purge
module load intel/19.0.5 openmpi/4.0.3
mpicc -qopenmp omp-hello.c -o omp-hello
salloc -t 00:30:00 -N 1 --exclusive srun -n 24 ./omp-hello
```
I get lots of `Hello World from thread number #`, no errors.

Reply:
I think -n 24 is wrong thing for threaded programs...for threaded programs you want 1 task with many threads that has access to many CPUs.

```
export OMP_NUM_THREADS=24
$ salloc -t 00:30:00 -N 1 --exclusive srun -n 1  ./omp-hello
```

That's the proper way.  Threads are not the same as CPUs so you can have 48 threads on 1 CPU so limiting srun or batch job to number of CPUs will not impact threads.  You will jiust end up with all threads on same core. Instead you want to use OMP_NUM_THREADS to control number of threads.